Test failures in pkg/storage/chunk/storage due to missing credentials

**Describe the bug**

Getting similar `google: could not find default credentials` errors to issue https://github.com/grafana/loki/issues/5832:

```
--- FAIL: TestChunksBasic (0.24s)
    --- FAIL: TestChunksBasic/bigtable-columnkey:true-gcsObjectClient:true-hashPrefix:true (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestChunksBasic/bigtable-columnkey:true-gcsObjectClient:true-hashPrefix:true
    --- FAIL: TestChunksBasic/bigtable-columnkey:true-gcsObjectClient:true-hashPrefix:false (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestChunksBasic/bigtable-columnkey:true-gcsObjectClient:true-hashPrefix:false
    --- FAIL: TestChunksBasic/bigtable-columnkey:false-gcsObjectClient:true-hashPrefix:true (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestChunksBasic/bigtable-columnkey:false-gcsObjectClient:true-hashPrefix:true
    --- FAIL: TestChunksBasic/bigtable-columnkey:false-gcsObjectClient:true-hashPrefix:false (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestChunksBasic/bigtable-columnkey:false-gcsObjectClient:true-hashPrefix:false
    --- FAIL: TestChunksBasic/caching-store (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestChunksBasic/caching-store
--- FAIL: TestIndexBasic (0.08s)
    --- FAIL: TestIndexBasic/bigtable-columnkey:true-gcsObjectClient:true-hashPrefix:true (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestIndexBasic/bigtable-columnkey:true-gcsObjectClient:true-hashPrefix:true
    --- FAIL: TestIndexBasic/bigtable-columnkey:true-gcsObjectClient:true-hashPrefix:false (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestIndexBasic/bigtable-columnkey:true-gcsObjectClient:true-hashPrefix:false
    --- FAIL: TestIndexBasic/bigtable-columnkey:false-gcsObjectClient:true-hashPrefix:true (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestIndexBasic/bigtable-columnkey:false-gcsObjectClient:true-hashPrefix:true
    --- FAIL: TestIndexBasic/bigtable-columnkey:false-gcsObjectClient:true-hashPrefix:false (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestIndexBasic/bigtable-columnkey:false-gcsObjectClient:true-hashPrefix:false
    --- FAIL: TestIndexBasic/caching-store (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestIndexBasic/caching-store
--- FAIL: TestQueryPages (0.04s)
    --- FAIL: TestQueryPages/bigtable-columnkey:true-gcsObjectClient:true-hashPrefix:true (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestQueryPages/bigtable-columnkey:true-gcsObjectClient:true-hashPrefix:true
    --- FAIL: TestQueryPages/bigtable-columnkey:true-gcsObjectClient:true-hashPrefix:false (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestQueryPages/bigtable-columnkey:true-gcsObjectClient:true-hashPrefix:false
    --- FAIL: TestQueryPages/bigtable-columnkey:false-gcsObjectClient:true-hashPrefix:true (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestQueryPages/bigtable-columnkey:false-gcsObjectClient:true-hashPrefix:true
    --- FAIL: TestQueryPages/bigtable-columnkey:false-gcsObjectClient:true-hashPrefix:false (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestQueryPages/bigtable-columnkey:false-gcsObjectClient:true-hashPrefix:false
    --- FAIL: TestQueryPages/caching-store (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestQueryPages/caching-store
--- FAIL: TestCardinalityLimit (0.03s)
    --- FAIL: TestCardinalityLimit/bigtable-columnkey:true-gcsObjectClient:true-hashPrefix:true (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestCardinalityLimit/bigtable-columnkey:true-gcsObjectClient:true-hashPrefix:true
    --- FAIL: TestCardinalityLimit/bigtable-columnkey:true-gcsObjectClient:true-hashPrefix:false (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestCardinalityLimit/bigtable-columnkey:true-gcsObjectClient:true-hashPrefix:false
    --- FAIL: TestCardinalityLimit/bigtable-columnkey:false-gcsObjectClient:true-hashPrefix:true (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestCardinalityLimit/bigtable-columnkey:false-gcsObjectClient:true-hashPrefix:true
    --- FAIL: TestCardinalityLimit/bigtable-columnkey:false-gcsObjectClient:true-hashPrefix:false (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestCardinalityLimit/bigtable-columnkey:false-gcsObjectClient:true-hashPrefix:false
    --- FAIL: TestCardinalityLimit/caching-store (0.00s)
        utils_test.go:34: 
            	Error Trace:	utils_test.go:34
            	Error:      	Received unexpected error:
            	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
            	Test:       	TestCardinalityLimit/caching-store
FAIL
FAIL	github.com/grafana/loki/pkg/storage/chunk/storage	0.733s
FAIL
```

**To Reproduce**
Using loki 2.5.0 sources, run `go test github.com/grafana/loki/pkg/storage/chunk/storage`.

**Expected behavior**
Tests passing without requiring any credentials.

**Environment:**
Arch Linux x86_64 while building loki 2.5.0 in a clean chroot

--- a/pkg/storage/chunk/client/gcp/fixtures.go
+++ b/pkg/storage/chunk/client/gcp/fixtures.go
@@ -83,7 +83,10 @@ func (f *fixture) Clients() (
 
 	if f.gcsObjectClient {
 		var c *GCSObjectClient
-		c, err = newGCSObjectClient(ctx, GCSConfig{BucketName: "chunks"}, hedging.Config{}, func(ctx context.Context, opts ...option.ClientOption) (*storage.Client, error) {
+		c, err = newGCSObjectClient(ctx, GCSConfig{
+			BucketName: "chunks",
+			Insecure:   true,
+		}, hedging.Config{}, func(ctx context.Context, opts ...option.ClientOption) (*storage.Client, error) {
 			return f.gcssrv.Client(), nil
 		})
 		if err != nil {

